server/monitor-types/mqtt.js

```diff
@@ -101,11 +101,9 @@ class MqttMonitorType extends MonitorType {
             });
 
             client.on("message", (messageTopic, message) => {
-                if (messageTopic === topic) {
-                    client.end();
-                    clearTimeout(timeoutID);
-                    resolve(message.toString("utf8"));
-                }
+                client.end();
```

Comment:
Would this not monitor on any channel instead?
Could you add a test case for this here: https://github.com/louislam/uptime-kuma/blob/master/test/backend-test/test-mqtt.js

Reply:
The client only subscribes to `topic`, so any other topic will be ignored, and we can be sure that `client.on("message")` is only triggered with the desired topic.

Good point about testing. I'll try to add test cases with wildcards and incorrect topics. If I understand the current tests correctly, `mqttTopic: "test"` is used all the time, and I'll need to change it.

Thanks for the feedback!

Reply:
I added a few tests, with different topics, including wildcard. Valid tests work fine.

I don't know enough about JS, async or promises in order to understand why those tests fail, though:

```
    test("invalid wildcard topic (with #)", async () => {
        await assert.rejects(
            await testMqtt("", null, "message", "#/c", "a/b/c"),
            new Error("Timeout, Message not received"),
        );
    });

    test("invalid wildcard topic (with +)", async () => {
        await assert.rejects(
            await testMqtt("", null, "message", "x/+/z", "a/b/c"),
            new Error("Timeout, Message not received"),
        );
    });

    test("invalid topic", async () => {
        await assert.rejects(
            await testMqtt("", null, "message", "x/+/z", "a/b/c"),
            new Error("Timeout, Message not received"),
        );
    });
```

The tests fail with `Error: Timeout, Message not received
      at Timeout._onTimeout (/home/ricou/Desktop/uptime-kuma/server/monitor-types/mqtt.js:70:24)`, which seems to be the Error we assert.

Do you understand why?

The monitor has the desired behaviour in the web-interface.

Reply:
> Do you understand why?

You are awaiting in the `assert.rejects`. If you remove this, it should work as far as I remember.

Reply:
Excellent, thanks!

All tests pass, including with invalid topics. The Pull Request should hopefully be good enough now.